interfaces/builtin/basedeclaration.go

```diff
@@ -178,6 +178,12 @@ slots:
         - core
         - gadget
     deny-auto-connection: true
+  boot-config:
+    allow-installation:
+      slot-snap-type:
+        - gadget
+    deny-connection: true
```

Comment:
@zyga the intent is that a snap declaration is required to use this interface and that intent is correct. However, as this PR is implemented it is a little confusing. Since the base declaration states this is *only* provided by a gadget snap, we would usually have some slot side policy and some plugs policy to access the slot side. As this PR is currently implemented, this feels more like it should be an implicit core slot (so slot-snap-type would change to 'core' and then we could 'deny-connection: true' up above in the 'plugs' section.

However, this interface isn't implemented how I expected. I expected:
1. the base declaration uses slot-snap-type: gadget and deny-installation: true (requires a snap declaration to install)
2. bootConfigPermanentSlotAppArmor contains write access to /boot/uboot/config.txt
3. the slot implementation supplies meta/hooks/configure
4. snapd is updated such that snaps that 'plugs: [ boot-config ]' may snap get/set the slot snap (that runs the configure hook). This requires that snapd checks the security label of the connecting process and sees if 'boot-config' is connected for the plugging snap and the snap it specified to snap get/set

Alternatively, snapd could instead handle all updates to /boot/uboot/config.txt itself such that neither the gadget nor the plugging snap has direct access to /boot/uboot/config.txt, the slot side (gadget) declares attributes of what can be set, and snapd mediates the access as in '4' above, but also rejects get/set for attributes not declared in the slot's attributes. IMHO, this is the best approach since snapd is the only thing that can modify /boot/uboot/config.txt and we can tightly control how that file is updated.